src/extract.py

```diff
@@ -1,36 +1,46 @@
 import praw
-from datetime import datetime
-#get client_id, client_secret, user_agent, username and password from praw.ini file
-#More information on configuring praw.ini files given here - https://praw.readthedocs.io/en/stable/getting_started/configuration/prawini.html
-reddit = praw.Reddit("DEFAULT") 
-
-# if reddit credentials are put in, below statement will be false since your reddit instance is not read only
-print(reddit.read_only)
-
-
-subreddit_name = "dataengineering"
-subreddit = reddit.subreddit(subreddit_name)
-
-n_of_posts = 0
-try: 
-    new_posts = subreddit.new(limit=10)
-    for post_id in new_posts: 
-        #n_of_posts = n_of_posts + 1
-        print(f"Post ID: {post_id}")
-        submission = reddit.submission(id=post_id)
-        print(f"# of comments: {submission.num_comments}")
-        
-        # Expand all comments and remove 'MoreComments' objects
-        submission.comments.replace_more(limit=None)
-        for comment in submission.comments.list(): 
-            print(f"Comment ID: {comment.id} Comment time: {datetime.utcfromtimestamp(comment.created_utc)}")
-        print("\n")
+import csv
+
+from datetime import datetime, timedelta
+from common.write_csv import write_post_data
+
+
+SUBMISSION_PULL_LIMIT = 50
+SUBREDDIT_NAME = "dataengineering"
+
+
+def main():
+
+    # Configure PRAW with credentials from praw.ini file
+    try:
+        reddit = praw.Reddit(config_file='praw.ini')
+    except (praw.exceptions.ClientException, FileNotFoundError) as e:
+        print(f"Error configuring PRAW: {e}")
+        return
+
+    subreddit = reddit.subreddit(SUBREDDIT_NAME)
+
+    yesterday_date = datetime.date(datetime.utcnow()) - timedelta(days=1)
+    
+    filename = "reddit-post-data-" + str(yesterday_date)
```

Comment:
Why do you need the yesterday_date? reddit api doesnt allow to fetch data from today's date?

Reply:
The idea is that if I run the code today, I will be able to pull all the submissions for yesterday's date.
This way, I don't have to worry if the DAG run is delayed by some time. As long as the DAG is run anytime today, I am sure to get the same submissions (i.e. all of yesterday's posts).